nodebuilder/core/config.go

```diff
@@ -42,11 +47,17 @@ func DefaultConfig() Config {
 			Port: DefaultPort,
 		},
 		AdditionalCoreEndpoints: make([]EndpointConfig, 0),
+		ConcurrencyLimit:        16,
 	}
 }
 
 // Validate performs basic validation of the config.
 func (cfg *Config) Validate() error {
+	if cfg.ConcurrencyLimit < core.DefaultConcurrencyLimit {
+		// override anything below the default limit
+		cfg.ConcurrencyLimit = core.DefaultConcurrencyLimit
+	}
```

Comment:
This is shady. We shouldn't silently override the configurations users explicitly set to be non-default. Like, if I set it 1 to like save resources, it would still blow up. 

We should error out if we don't allow setting it below the default, or only dissalow setting 0.

Reply:
@Wondertan I agree it's "shady" to override but it's not breaking out current behaviour. Currently we just spin up 16 routines and our resource requirements for BNs are sufficient to cover that. Are you saying we should allow less now?

Reply:
No, I don't mind restricting it to be more than 16 only if you think this is better, but by erroring out and not overriding

Reply:
Sure 🤝